GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/MountTests.cs

```diff
@@ -210,6 +210,21 @@ public void MountRegeneratesAlternatesFileWhenMissingFromDisk()
             alternatesFilePath.ShouldBeAFile(this.fileSystem).WithContents(objectsRoot);
         }
 
+        [TestCase]
+        public void MountCanProcessSavedBackgroundQueueTasks()
+        {
+            this.Enlistment.UnmountGVFS();
+
+            string deleteFilePath = Path.Combine("Test_EPF_WorkingDirectoryTests", "1", "2", "3", "4", "ReadDeepProjectedFile.cpp");
+            string deleteDirPath = Path.Combine("Test_EPF_WorkingDirectoryTests", "1", "2", "3", "4");
+            this.fileSystem.WriteAllText(
+                Path.Combine(this.Enlistment.EnlistmentRoot, ".gvfs", "databases", "BackgroundGitOperations.dat"),
+                "A 1\03\0{deleteFilePath}\0\r\nA 2\011\0{deleteDirPath}\0\r\n");

```

Comment:
This test should behave a little different on Mac and Windows.

On Windows we receive/persist the `OnFileDeleted` and `OnFolderDeleted` notifications (3 and 11), but on Mac we receive/persist `OnFilePreDelete` and `OnFolderPreDelete` (16 and 17).

Reply:
@wilbaker , in that case how would you recommend we handle it.  Would a switch like we did here for https://github.com/Microsoft/VFSForGit/blob/master/GVFS/GVFS.FunctionalTests/Tests/DiskLayoutVersionTests.cs, for OSPlatform.OSX be most appropriate?